packages/server-core/src/media/storageprovider/s3.storage.ts

```diff
@@ -281,7 +299,7 @@ export class S3Provider implements StorageProviderInterface {
    */
   async deleteResources(keys: string[]) {
     // Create batches of 1000 since S3 supports deletion of 1000 object max per request
-    const batches = [] as ObjectIdentifierList[]
+    const batches: any = [] as ObjectIdentifier
```

Comment:
It seems like `ObjectIdentifier` is not of type array. Also `:any` should not be used.
I guess following would be correct statement:
```const batches = [] as ObjectIdentifier[]```


https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/modules/objectidentifier.html

Reply:
So, here the issue I was facing was push does not exists on ObjectIdentifier. The ObjectIdentifierList is not in the version 3 sdk. I was referring this: https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/classes/deleteobjectscommand.html . And also the Key parameter is mendatory in ObjectIdentifier.

Reply:
Correction, I think you need to use:
```
const batches = [] as ObjectIdentifier[][]
```

Reply:
I updated the script.